script/model/forecast.py

```diff
@@ -0,0 +1,520 @@
+#!/usr/bin/env python3
+"""
+TODO(ricky):
+The modeling scripts should be under a project, with __init__
+What should be the best dependency between the modeling scripts and testing package?
+"""
+import sys
+from pathlib import Path
+sys.path.insert(0, str((Path.cwd() / '..' / 'testing').absolute()))
+
+from oltpbench.test_case_oltp import TestCaseOLTPBench
+from oltpbench.run_oltpbench import TestOLTPBench
+from oltpbench.constants import OLTPBENCH_GIT_LOCAL_PATH
+from util.constants import LOG, ErrorCode
+from util.common import run_command
+from util.db_server import NoisePageServer
+from xml.etree import ElementTree
+from typing import Dict, List, Optional, Tuple
+import torch.nn as nn
+import torch
+from sklearn.preprocessing import MinMaxScaler
+import numpy as np
+import matplotlib.pyplot as plt
+import pickle
+import logging
+import csv
+import argparse
+
+
+# Default pattern: High -> Low  ...
+DEFAULT_TPCC_RATE = 10000
+DEFAULT_WORKLOAD_PATTERN = [DEFAULT_TPCC_RATE, DEFAULT_TPCC_RATE // 10]
+
+# Default time, runs 30 second for a work phase
+DEFAULT_TPCC_TIME_SEC = 30
+
+# Run the workload pattern for 30 iterations
+DEFAULT_ITER_NUM = 1
+
+# Load the workload pattern - based on the tpcc.json in
+# testing/oltpbench/config
+DEFAULT_TPCC_WEIGHTS = "45,43,4,4,4"
+DEFAULT_OLTP_TEST_CASE = {
+    "benchmark": "tpcc",
+    "query_mode": "extended",
+    "terminals": 4,
+    "scale_factor": 4,
+    "weights": DEFAULT_TPCC_WEIGHTS
+}
+
+# Enable query trace collection, it will produce a query_trace.csv at CWD
+DEFAULT_OLTP_SERVER_ARGS = {
+    "server_args": {
+        'query_trace_metrics_enable': None
+    }
+}
+
+# Default query_trace file name
+DEFAULT_QUERY_TRACE_FILE = "query_trace.csv"
+
+# Interval duration for aggregation in MS
+INTERVAL_MS = 500000
+
+# Number of MS per second
+MS_PER_SEC = 1000000
+
+# First 35 seconds are for loading
+LOAD_PHASE_NUM_DATA = int(MS_PER_SEC / INTERVAL_MS * 35)
+
+# Number of data points in a sequence
+SEQ_LEN = 10 * MS_PER_SEC // INTERVAL_MS
+
+# Number of data points for the horizon
+HORIZON_LEN = 30 * MS_PER_SEC // INTERVAL_MS
+
+# Number of data points for testing set
+EVAL_DATA_SIZE = 2 * SEQ_LEN + HORIZON_LEN
+
+
+def plot_ts(series: List[Tuple], interval_sec: float) -> None:
+    """
+    Utility function for plotting
+    :param series: List of tuple of (Label:str, X:List[int], Data:List[Numerical]):
+        e.g. [("Train", x_labels, data)]
+    :param interval_sec: Interval seconds (used only in descriptive manner)
+    :return:
+    """
+    plt.title(f'Time series Query Count per {interval_sec} seconds')
+    plt.ylabel(f'Query Count / {interval_sec}')
+    plt.xlabel('Time')
+    plt.grid(True)
+    plt.autoscale(axis='x', tight=True)
+
+    for serie in series:
+        label, x, data = serie
+        plt.plot(x, data, label=label)
+    plt.legend()
+
+
+def gen_forecast_data(xml_config_file: str, rate_pattern: List[int]) -> None:
```

Comment:
This is just setting up the config file instead of actually generating the data. So I would rename this function.